### PR TITLE
CI: simplify and improve name of docs build

### PR DESCRIPTION
**Pull request by tacaswell**

### CODE REVIEW

**Comment by prjemian** [comment]
Is a matrix build necessary now?

**Comment by tacaswell** [comment]
no, this job is just building the docs and I do not think we want to test that sphinx works on all versions of Python.

**Comment by prjemian** [comment]
So additional change would be to remove the matrix from the workflow file?

**Comment by prjemian** [comment]
These places:

https://github.com/nexusformat/definitions/blob/6597e7f6c6a73df93aa9448fee2cc69669c9d846/.github/workflows/ci.yaml#L29
https://github.com/nexusformat/definitions/blob/6597e7f6c6a73df93aa9448fee2cc69669c9d846/.github/workflows/ci.yaml#L45
https://github.com/nexusformat/definitions/blob/6597e7f6c6a73df93aa9448fee2cc69669c9d846/.github/workflows/ci.yaml#L48

remove this section:
https://github.com/nexusformat/definitions/blob/6597e7f6c6a73df93aa9448fee2cc69669c9d846/.github/workflows/ci.yaml#L31-L34

And while we're at it, change the file name from `ci.yml` to something more precise, such as `docs.yml`.

File: .github/workflows/validate.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        run: |
+          python -m dev_tools nxtest
+
+      - name: cnxvalidate build deps
+        run: |
+          sudo apt-get update -y && \
+          sudo apt-get install -y \
+             build-essential libhdf5-dev
+
+      - name: install cnxvalidate
```
Comment (prjemian) [inline .github/workflows/validate.yaml:47]: Not appropriate for the definitions repository.  This belongs in the cnxvalidate repo.

Comment (tacaswell) [reply .github/workflows/validate.yaml:47]: This is step 1 of a plan.

**Comment by prjemian** [comment]
Can the additions of [0ecac0d](https://github.com/nexusformat/definitions/pull/1473/commits/0ecac0d13d0adbe354929f561a213504138e738f) be part of a different PR?  They expand the narrow scope of this PR, to "simplify and improve name".

Isn't validation of the NXDL files already happening, via the XML tools?  This code? https://github.com/nexusformat/definitions/blob/2ee18883149f88b8c2aa9f87b8804a73b1c79f54/dev_tools/nxdl/syntax.py#L15-L25

**Comment by prjemian** [comment]
Release of the definitions should not become dependent on first revision of any of the validation repositories.  This quickly becomes a deadlock.

File: .github/workflows/validate.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+          python-version: "3.12"
+
+      - name: Install Requirements
+        run: |
+          python -m pip install --upgrade pip setuptools
+          python -m pip install -r requirements.txt
+
+      - name: validate
+        run: |
+          python -m dev_tools nxtest
```
Comment (prjemian) [inline .github/workflows/validate.yaml:39]: Does this check **all** NXDL files or just the nxtest.nxdl.xml file?

Comment (tacaswell) [reply .github/workflows/validate.yaml:39]: all.

Comment (prjemian) [reply .github/workflows/validate.yaml:39]: Can additional logging be enabled that shows each NXDL file as it is being validated?  Not see this in the recent validation CI logs:
https://productionresultssa18.blob.core.windows.net/actions-results/a8a7d118-b555-4269-8bbd-a422a596a1d6/workflow-job-run-d5c11ce9-a08a-5bde-783f-1bfee628ff52/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-09-28T12%3A29%3A02Z&sig=%2B%2BsuRQU%2Bi1l4%2FTda7NTubMnBen85z2Y%2FWxVvNVLWgW4%3D&ske=2024-09-28T23%3A30%3A09Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-09-28T11%3A30%3A09Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2024-08-04&sp=r&spr=https&sr=b&st=2024-09-28T12%3A18%3A57Z&sv=2024-08-04

**Comment by prjemian** [comment]
Not seeing these prints in the latest [log](https://productionresultssa18.blob.core.windows.net/actions-results/a8a7d118-b555-4269-8bbd-a422a596a1d6/workflow-job-run-d5c11ce9-a08a-5bde-783f-1bfee628ff52/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-09-28T12%3A39%3A44Z&sig=4MGSQHKgYzmmw8dx2hCkyl%2BD05nXNCqxeEDPtC4zvDo%3D&ske=2024-09-28T23%3A23%3A48Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-09-28T11%3A23%3A48Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2024-08-04&sp=r&spr=https&sr=b&st=2024-09-28T12%3A29%3A39Z&sv=2024-08-04).

File: .github/workflows/validate.yaml (hunk 2 above)
Comment (prjemian) [reply .github/workflows/validate.yaml:39]: Looking at the [Quick Reference](https://github.com/nexusformat/definitions/tree/main/dev_tools#quick-reference), I'm looking for:

`pytest dev_tools`

Comment (prjemian) [reply .github/workflows/validate.yaml:39]: Locally, this command will test all NXDL files.

Comment (prjemian) [reply .github/workflows/validate.yaml:39]: If we stopped validating NXDL files on every push, that was a regression.

Comment (prjemian) [reply .github/workflows/validate.yaml:39]:
```bash
(nexus_defs) prjemian@arf:~/.../NeXus/definitions$ python -m dev_tools nxtest
usage: __main__.py [-h] {nxclass,manual,impatient} ...
__main__.py: error: argument command: invalid choice: 'nxtest' (choose from 'nxclass', 'manual', 'impatient')
```

Comment (prjemian) [reply .github/workflows/validate.yaml:39]: OK, tested with your branch and confirm this tests all existing NXDL files.  It reported properly an error that I introduced into my local NXdata just to test the code.

**Review by prjemian — CHANGES_REQUESTED** [review]
I like that validation is done in a separate workflow.  Can we run it every push, not just on push to main?  That will catch problems at their onset.  History on this repo has proven the value of that early feedback.  Y'all did say: _Fail early, fail hard._

**Review by prjemian — APPROVED** [review]
OK, Thanks!

**Comment by prjemian** [comment]
@tacaswell Take the glory and merge this approved PR.  Is it your first here with NeXus?

**Pull request merged**